packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js

```diff
@@ -180,15 +180,29 @@ if (__DEV__) {
       return true;
     }
 
-    if (typeof value === 'boolean') {
-      warning(
-        DOMProperty.shouldAttributeAcceptBooleanValue(name),
-        'Received `%s` for non-boolean attribute `%s`. If this is expected, cast ' +
-          'the value to a string.%s',
-        value,
-        name,
-        getStackAddendum(),
-      );
+    if (
+      typeof value === 'boolean' &&
+      !DOMProperty.shouldAttributeAcceptBooleanValue(name)
+    ) {
+      if (value === true) {
+        warning(
+          false,
+          'Received `%s` for non-boolean attribute `%s`. If this is expected, cast ' +
+            'the value to a string.%s',
+          value,
+          name,
+          getStackAddendum(),
+        );
+      } else {
+        warning(
+          false,
+          'If you mean to conditionally pass an attribute, use a ternary ' +
```

Comment:
Can this also start with `Received ... for non-boolean attribute ...`? I want both messages to look similar, but with different second sentences.